R/parameters.R

```diff
@@ -370,3 +382,76 @@ beta_est_infectiousness <- function(dur_IMild,
   R0 / adjusted_eigen
 
 }
+
+#' @noRd
+format_ve_i_for_odin <- function(vaccine_efficacy_infection,
+                                 tt_vaccine_efficacy_infection) {
+
+# If just provided as a vector then we put into a list ready for formatting
+if(!is.list(vaccine_efficacy_infection)){
+  vaccine_efficacy_infection <- list(vaccine_efficacy_infection)
+}
+
+# check that the correct length agreement between tt_vaccine_efficacy_infection
+assert_length(vaccine_efficacy_infection, length(tt_vaccine_efficacy_infection))
+
+# and now format so each list is the vaccine_efficacy_infection at each time
+# point for the 6 vaccine classes
+ve_i_list <- lapply(vaccine_efficacy_infection, function(ve_i) {
+
+  ve_i = 1 - ve_i
+  return(matrix(c(rep(1, 17 * 3),
+                  ve_i, ve_i,
+                  rep(1, 17)), nrow = 17, ncol = 6))
+
```

Comment:
In the previous PR to develop (https://github.com/mrc-ide/nimue/pull/39) we made changes so that the user could either specific 1 single efficacy value, or a vector of length 17. These would then be resolved in the parmeters call:
https://github.com/mrc-ide/nimue/blob/9228c4115119f905009b40415c9852b5dbaa588f/R/parameters.R#L258-L274

I think this reverts back to only allowing vector (or list of vectors) of length 17. That's fine, but if so we should have a check on inputs (length ==17, or length of all levels of list ==7) here otherwise this can lead to weird results if a single value, not a vector is provided. Same goes for both efficacy inputs I think.

Reply:
Hey, not sure if these commits came in after you reviewed, but we pick this up here https://github.com/mrc-ide/nimue/pull/41/files#diff-137d718505d68eb93303519acecb5c7dc7895c1aa4b4b6a4c92dec39fd8e9953R400-R413 to make sure that if any list element is length 1 then we rep it out to 17 and then do the length check after as you put in

Reply:
Ahh ok, must have missed that somehow 👍